chore(samples): Code samples for Retail Tutorials. Search service samples

t-karasova opened the pull request:
This reverts commit 153c92d2

moved to interactive-tutorials dir

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-retail/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

t-karasova committed:
feat: Code samples for Retail Tutorials. Search service samples

This reverts commit 153c92d2

moved to interactive-tutorials dir

Fix lint checkstyle comments

Fix unit test

t-karasova committed:
feat: Code samples for Retail Tutorials. Search service samples

t-karasova committed:
feat: Code samples for Retail Tutorials. Search service samples

This reverts commit 153c92d2

moved to interactive-tutorials dir

Commit:
Revert "feat: Code samples for Retail Tutorials. Search service samples"

This reverts commit 153c92d2

renovate-bot committed:
build(deps): update dependency com.google.cloud:google-cloud-shared-config to v1.2.5 (#265)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-config](https://togithub.com/googleapis/java-shared-config) | `1.2.4` -> `1.2.5` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.5/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.5/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.5/compatibility-slim/1.2.4)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.5/confidence-slim/1.2.4)](https://docs.renovatebot.com/merge-confidence/) |

---

<details>
<summary>googleapis/java-shared-config</summary>

[Compare Source](https://togithub.com/googleapis/java-shared-config/compare/v1.2.4...v1.2.5)

</details>

---

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Renovate will not automatically rebase this PR, because other commits have been found.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

Neenu1995 committed:
chore: update release-level in .repo-metadata.json (#268)

renovate-bot committed:
chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.7 (#261)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-retail](https://togithub.com/googleapis/java-) | `2.0.6` -> `2.0.7` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-retail/2.0.7/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-retail/2.0.7/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-retail/2.0.7/compatibility-slim/2.0.6)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-retail/2.0.7/confidence-slim/2.0.6)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Renovate will not automatically rebase this PR, because other commits have been found.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

gcf-owl-bot[bot] committed:
chore(java): update release_type choices to stable and preview (#1331) (#269)

t-karasova commented:
@kurtisvg
Hi Kurtis,
I have created this pull request to add the code samples for Interactive Tutorials for Retail API.
You have already seen them in [#254](https://github.com/googleapis/java-retail/pull/254), which now is closed, as it was too big.

The code samples are moved to the folder "samples/interactive-tutorials" to separate them from the code samples, which are used for displaying in the static documentation pages.
It was decided, that our tutorials + code samples will be shown only by clicking the button "Guide me" in the Retail documentation page and will be opened in the Cloud Shell IDE.

In this PR I have created the code samples for Retail Search service and tests.

I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?

Thank you

sborisenkox committed:
Fix lint checkstyle comments

sborisenkox committed:
Fix unit test

renovate-bot committed:
build(deps): update dependency com.google.cloud:google-cloud-shared-config to v1.2.6 (#270)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-config](https://togithub.com/googleapis/java-shared-config) | `1.2.5` -> `1.2.6` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.6/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.6/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.6/compatibility-slim/1.2.5)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.6/confidence-slim/1.2.5)](https://docs.renovatebot.com/merge-confidence/) |

---

### Release Notes

<details>
<summary>googleapis/java-shared-config</summary>

### [`v1.2.6`](https://togithub.com/googleapis/java-shared-config/blob/HEAD/CHANGELOG.md#&#8203;126-httpsgithubcomgoogleapisjava-shared-configcomparev125v126-2022-01-19)

[Compare Source](https://togithub.com/googleapis/java-shared-config/compare/v1.2.5...v1.2.6)

</details>

---

### Configuration

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

chenlei1216 commented on samples/interactive-tutorials/src/main/java/search/SearchWithOrdering.java:
```diff
@@ -0,0 +1,118 @@
+   * @throws IOException if endpoint is not provided.
+   */
+  public static SearchResponse search() throws IOException {
+    // TRY DIFFERENT FILTER EXPRESSIONS HERE:
```
ORDER BY EXPRESSIONS?

t-karasova replied:
this file is removed to make the PR smaller - will be submitted in the following PR

chenlei1216 commented on samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java:
```diff
@@ -0,0 +1,127 @@
+
+    System.out.println("Search response: " + searchResponseFirstPage);
+
+    // PASTE CALL WITH NEXT PAGE TOKEN HERE:
```
Maybe add code samples to get the next page token?

t-karasova replied:
We provide a small piece of code to get the nextPageToken and use it the similar search request in the tutorial, ask user to copy and paste the code here

t-karasova replied:
this file is removed to make the PR smaller - will be submitted in the following PR

t-karasova committed:
smal fix

t-karasova committed:
secter manager retail-samples-secrets is added

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

sborisenkox committed:
Merge branch 'googleapis:main' into main

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-shared-dependencies to v2.7.0 (#275)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-dependencies](https://togithub.com/googleapis/java-shared-dependencies) | `2.6.0` -> `2.7.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.7.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.7.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.7.0/compatibility-slim/2.6.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.7.0/confidence-slim/2.6.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Release Notes

<details>
<summary>googleapis/java-shared-dependencies</summary>

### [`v2.7.0`](https://togithub.com/googleapis/java-shared-dependencies/blob/HEAD/CHANGELOG.md#&#8203;270-httpsgithubcomgoogleapisjava-shared-dependenciescomparev260v270-2022-01-28)

[Compare Source](https://togithub.com/googleapis/java-shared-dependencies/compare/v2.6.0...v2.7.0)

##### Dependencies

-   update dependency com.fasterxml.jackson:jackson-bom to v2.13.1 ([#&#8203;555](https://togithub.com/googleapis/java-shared-dependencies/issues/555)) ([67b5663](https://togithub.com/googleapis/java-shared-dependencies/commit/67b56635509215fd132b2ffbcaed995a6de93879))
-   update dependency com.google.api-client:google-api-client-bom to v1.33.1 ([#&#8203;580](https://togithub.com/googleapis/java-shared-dependencies/issues/580)) ([895dba2](https://togithub.com/googleapis/java-shared-dependencies/commit/895dba2afef6c295b0e21a4ccd27feee1b6af8e2))
-   update dependency com.google.api:api-common to v2.1.3 ([#&#8203;587](https://togithub.com/googleapis/java-shared-dependencies/issues/587)) ([5dbbce8](https://togithub.com/googleapis/java-shared-dependencies/commit/5dbbce80baa045818757912e6db4f222c12b75f6))
-   update dependency com.google.auth:google-auth-library-bom to v1.4.0 ([#&#8203;578](https://togithub.com/googleapis/java-shared-dependencies/issues/578)) ([c274c62](https://togithub.com/googleapis/java-shared-dependencies/commit/c274c62905192bcc48c82104bf50cbeaa1894c2b))
-   update dependency com.google.errorprone:error_prone_annotations to v2.11.0 ([#&#8203;582](https://togithub.com/googleapis/java-shared-dependencies/issues/582)) ([314da38](https://togithub.com/googleapis/java-shared-dependencies/commit/314da38c9f12cf425b5d1be8297e9f2a78d13948))
-   update dependency com.google.http-client:google-http-client-bom to v1.41.2 ([#&#8203;579](https://togithub.com/googleapis/java-shared-dependencies/issues/579)) ([c8df06d](https://togithub.com/googleapis/java-shared-dependencies/commit/c8df06ddb3f4a975e928bbdd9dfae293e81efa39))
-   update dependency com.google.oauth-client:google-oauth-client-bom to v1.33.0 ([#&#8203;576](https://togithub.com/googleapis/java-shared-dependencies/issues/576)) ([3315988](https://togithub.com/googleapis/java-shared-dependencies/commit/3315988059ca0223f078b0f760960674bfa8c04a))
-   update dependency com.google.protobuf:protobuf-bom to v3.19.3 ([#&#8203;571](https://togithub.com/googleapis/java-shared-dependencies/issues/571)) ([c1931e1](https://togithub.com/googleapis/java-shared-dependencies/commit/c1931e1c4611ca3d1e2afac1c9e983cf1ec4a7de))
-   update dependency io.grpc:grpc-bom to v1.44.0 ([#&#8203;586](https://togithub.com/googleapis/java-shared-dependencies/issues/586)) ([52c47b7](https://togithub.com/googleapis/java-shared-dependencies/commit/52c47b7425af438ba1a82dac87f9002353a6a87a))
-   update dependency org.checkerframework:checker-qual to v3.21.1 ([#&#8203;554](https://togithub.com/googleapis/java-shared-dependencies/issues/554)) ([81bbb91](https://togithub.com/googleapis/java-shared-dependencies/commit/81bbb91774ab67161371bfd0dfd143de89dee4c9))
-   update gax.version to v2.11.0 ([#&#8203;581](https://togithub.com/googleapis/java-shared-dependencies/issues/581)) ([1899612](https://togithub.com/googleapis/java-shared-dependencies/commit/189961226f84cdd4e283780d2c7d4a4dffc3e00c))
-   update gax.version to v2.9.0 ([#&#8203;573](https://togithub.com/googleapis/java-shared-dependencies/issues/573)) ([ee5e38e](https://togithub.com/googleapis/java-shared-dependencies/commit/ee5e38efea574eeb2b5803192da98ba1ce1c8e6e))
-   update google.common-protos.version to v2.7.2 ([#&#8203;588](https://togithub.com/googleapis/java-shared-dependencies/issues/588)) ([caac0e6](https://togithub.com/googleapis/java-shared-dependencies/commit/caac0e68e336b801becb6322b5b81fce1824ad8a))
-   update google.core.version to v2.4.0 ([#&#8203;589](https://togithub.com/googleapis/java-shared-dependencies/issues/589)) ([6a292e3](https://togithub.com/googleapis/java-shared-dependencies/commit/6a292e3530a06d6b228f3154ed4e979ef68d7aeb))
-   update iam.version to v1.2.1 ([#&#8203;569](https://togithub.com/googleapis/java-shared-dependencies/issues/569)) ([62dd270](https://togithub.com/googleapis/java-shared-dependencies/commit/62dd2705b257c4bb539bfe619234f38d0e13b138))
-   update opencensus.version to v0.30.0 ([#&#8203;552](https://togithub.com/googleapis/java-shared-dependencies/issues/552)) ([ebae62e](https://togithub.com/googleapis/java-shared-dependencies/commit/ebae62eca75234db4fd571f42b8ed0f6ce1bd1f7))
-   update opencensus.version to v0.31.0 ([#&#8203;585](https://togithub.com/googleapis/java-shared-dependencies/issues/585)) ([021b066](https://togithub.com/googleapis/java-shared-dependencies/commit/021b066a833af1629c89fcca593d16b9786ee0bf))

</details>

---

### Configuration

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,115 @@
+ */
+
+/*
+ * [START retail_search_for_products_with_query_parameter]
```
The region tag should be in a stand alone comment, otherwise the comment below will not render correctly on the page.

t-karasova replied:
fixed

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,115 @@
+
+public final class SearchSimpleQuery {
+
+  /** This variable describes project number getting from environment variable. */
+  private static final String YOUR_PROJECT_NUMBER = System.getenv("PROJECT_NUMBER");
+
+  /** This variable describes default catalog name. */
+  private static final String DEFAULT_CATALOG_NAME =
+      String.format("projects/%s/locations/global/catalogs/default_catalog", YOUR_PROJECT_NUMBER);
+
+  /**
+   * This variable describes default search placement name. Using for identify the Serving Config
+   * name.
+   */
+  private static final String DEFAULT_SEARCH_PLACEMENT_NAME =
+      DEFAULT_CATALOG_NAME + "/placements/default_search";
+
+  /** This variable describes a unique identifier to track visitors. */
+  private static final String VISITOR_ID = UUID.randomUUID().toString();
```
This class should begin with a main function, which provides examples for the user to override to set any required variables. Env vars can be used from the tests to trigger the function itself. 

Please see https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#method-structure.

t-karasova replied:
done

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,115 @@
+  /** This variable describes a unique identifier to track visitors. */
+  private static final String VISITOR_ID = UUID.randomUUID().toString();
+
+  private SearchSimpleQuery() {}
```
Why is this here?

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,115 @@
+   * @return SearchServiceClient.
+   * @throws IOException if endpoint is incorrect.
+   */
+  private static SearchServiceClient getSearchServiceClient() throws IOException {
```
Why is this in a separate function? 

Please see https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#client-initialization

t-karasova replied:
done

kurtisvg reviewed (changes requested):
I reviewed a single file for now. This PR is exceptionally large (+1500) lines. 

Please see go/small-cls and break it into smaller PRs so it's easier to review.

kurtisvg commented:
> I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

I don't see any modifications to kokoro in these config. Typically we don't modify the kokoro configs as they are often generated by OwlBot. @lesv or @Neenu1995 might be able to provide more info on what the preferred paths for env vars is here

> For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?

Maybe @lesv or @Neenu1995 can weigh in on this too. I think the rest of the java tests use a single project these days, but I'm not sure.

Neenu1995 commented:
> I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

I don't see the kokoro configs either.

> For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?

The secret key needs to be rotated periodically. It can be done using the [secret rotator](https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/guides/using-secret-rotator-service.md?cl=head). It will be easier to use the existing project to run tests unless these tests can interfere with existing test resources.

Neenu1995 commented:
Let us know which method you decide to use and we can work together on the infrastructure to set that up.
Also in the existing configuration, continuous tests have not been set up in kokoro. You will have to enable it if you plan to use continuous scheduled tests instead of presubmit test.

gcf-owl-bot[bot] committed:
deps(java): update actions/github-script action to v5 (#1339) (#272)

sborisenkox committed:
Merge remote-tracking branch 'origin/crs-tutorials-search' into crs-tutorials-search

# Conflicts:
#	samples/interactive-tutorials/src/main/java/search/SearchWithOrdering.java

t-karasova committed:
Code changes.

t-karasova committed:
kokoro config

t-karasova committed:
kokoro config

t-karasova committed:
kokoro config

t-karasova committed:
kokoro config

gcf-owl-bot[bot] committed:
chore: add kokoro config to run nightly integration test on java11.0.14 (#1347) (#278)

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

release-please[bot] committed:
chore(main): release 2.0.8 (#276)

:robot: I have created a release *beep* *boop*
---


### [2.0.8](https://github.com/googleapis/java-retail/compare/v2.0.7...v2.0.8) (2022-02-03)


### Dependencies

* **java:** update actions/github-script action to v5 ([#1339](https://github.com/googleapis/java-retail/issues/1339)) ([#272](https://github.com/googleapis/java-retail/issues/272)) ([14ba816](https://github.com/googleapis/java-retail/commit/14ba816df244116263397c8da9b2a18b4c7f43d3))
* update dependency com.google.cloud:google-cloud-shared-dependencies to v2.7.0 ([#275](https://github.com/googleapis/java-retail/issues/275)) ([51a2492](https://github.com/googleapis/java-retail/commit/51a2492e4b862af408999ff5473984299e0b68a1))

---
This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).

release-please[bot] committed:
chore(main): release 2.0.9-SNAPSHOT (#279)

:robot: I have created a release *beep* *boop*
---


### Updating meta-information for bleeding-edge SNAPSHOT release.

---
This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).

t-karasova committed:
Job_type added to build.sh

feat: Code samples for Retail Tutorials. Search service samples

Revert "feat: Code samples for Retail Tutorials. Search service samples"

This reverts commit 153c92d2

build(deps): update dependency com.google.cloud:google-cloud-shared-config to v1.2.4 (#257)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-config](https://togithub.com/googleapis/java-shared-config) | `1.2.2` -> `1.2.4` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/compatibility-slim/1.2.2)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/confidence-slim/1.2.2)](https://docs.renovatebot.com/merge-confidence/) |

---

<details>
<summary>googleapis/java-shared-config</summary>

[Compare Source](https://togithub.com/googleapis/java-shared-config/compare/v1.2.3...v1.2.4)

[Compare Source](https://togithub.com/googleapis/java-shared-config/compare/v1.2.2...v1.2.3)

</details>

---

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

deps: update dependency com.google.cloud:google-cloud-shared-dependencies to v2.6.0 (#258)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-dependencies](https://togithub.com/googleapis/java-shared-dependencies) | `2.5.1` -> `2.6.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/compatibility-slim/2.5.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/confidence-slim/2.5.1)](https://docs.renovatebot.com/merge-confidence/) |

---

<details>
<summary>googleapis/java-shared-dependencies</summary>

[Compare Source](https://togithub.com/googleapis/java-shared-dependencies/compare/v2.5.1...v2.6.0)

-   **java:** add -ntp flag to native image testing command ([#&#8203;1299](https://www.togithub.com/googleapis/java-shared-dependencies/issues/1299)) ([#&#8203;546](https://www.togithub.com/googleapis/java-shared-dependencies/issues/546)) ([d8bb8d9](https://www.github.com/googleapis/java-shared-dependencies/commit/d8bb8d9d8e3b8c582f768cd3164e8580ec334511))
-   **java:** run Maven in plain console-friendly mode ([#&#8203;1301](https://www.togithub.com/googleapis/java-shared-dependencies/issues/1301)) ([#&#8203;557](https://www.togithub.com/googleapis/java-shared-dependencies/issues/557)) ([6b57d08](https://www.github.com/googleapis/java-shared-dependencies/commit/6b57d08816ec6c1dfb637cae022624e896930ad8))

-   gRPC upgrade to 1.43.2 ([#&#8203;567](https://www.togithub.com/googleapis/java-shared-dependencies/issues/567)) ([c946655](https://www.github.com/googleapis/java-shared-dependencies/commit/c94665571f26d24fbbb3f89ef5a3db99663bfdba))
-   update dependency com.google.api-client:google-api-client-bom to v1.33.0 ([#&#8203;563](https://www.togithub.com/googleapis/java-shared-dependencies/issues/563)) ([0aedddb](https://www.github.com/googleapis/java-shared-dependencies/commit/0aedddb9a535753767d56f5910f355c7a1c8509c))
-   update dependency com.google.api:api-common to v2.1.2 ([2f154a3](https://www.github.com/googleapis/java-shared-dependencies/commit/2f154a3ff759211438c72859e98b314c7ff8405f))
-   update dependency com.google.http-client:google-http-client-bom to v1.41.0 ([#&#8203;562](https://www.togithub.com/googleapis/java-shared-dependencies/issues/562)) ([6da0735](https://www.github.com/googleapis/java-shared-dependencies/commit/6da0735d4db9df998d11c549488a024b800bed65))
-   update dependency io.grpc:grpc-bom to v1.43.1 ([#&#8203;553](https://www.togithub.com/googleapis/java-shared-dependencies/issues/553)) ([b787c8f](https://www.github.com/googleapis/java-shared-dependencies/commit/b787c8f2032b7ffc9797158c3ff214b7813f34b7))
-   update dependency org.apache.httpcomponents:httpcore to v4.4.15 ([#&#8203;550](https://www.togithub.com/googleapis/java-shared-dependencies/issues/550)) ([e2d3dba](https://www.github.com/googleapis/java-shared-dependencies/commit/e2d3dbad6dd54c117abb5ab1b6137cfee4e5a32b))
-   update dependency org.checkerframework:checker-qual to v3.20.0 ([#&#8203;549](https://www.togithub.com/googleapis/java-shared-dependencies/issues/549)) ([586df59](https://www.github.com/googleapis/java-shared-dependencies/commit/586df5949d4ddca11d0705d696882c32d8436c31))
-   update gax.version to v2.8.0 ([#&#8203;564](https://www.togithub.com/googleapis/java-shared-dependencies/issues/564)) ([9ad5d4f](https://www.github.com/googleapis/java-shared-dependencies/commit/9ad5d4f5385753a0c5739ef400f5d346e3342b84))
-   update gax.version to v2.8.1 ([2f154a3](https://www.github.com/googleapis/java-shared-dependencies/commit/2f154a3ff759211438c72859e98b314c7ff8405f))
-   update google.common-protos.version to v2.7.1 ([#&#8203;561](https://www.togithub.com/googleapis/java-shared-dependencies/issues/561)) ([9aa860d](https://www.github.com/googleapis/java-shared-dependencies/commit/9aa860d92f745f1d4382eb16439bf17ac6ca8ae9))
-   update google.core.version to v2.3.5 ([#&#8203;565](https://www.togithub.com/googleapis/java-shared-dependencies/issues/565)) ([2f154a3](https://www.github.com/googleapis/java-shared-dependencies/commit/2f154a3ff759211438c72859e98b314c7ff8405f))
-   update iam.version to v1.2.0 ([#&#8203;547](https://www.togithub.com/googleapis/java-shared-dependencies/issues/547)) ([3232c82](https://www.github.com/googleapis/java-shared-dependencies/commit/3232c828b628cb604a200b3e569e56e758e5e11d))
-   upgrading protobuf to 3.19.2 ([#&#8203;560](https://www.togithub.com/googleapis/java-shared-dependencies/issues/560)) ([7ef3325](https://www.github.com/googleapis/java-shared-dependencies/commit/7ef33250f5473348d3546210b7a534360f476101))

-   update dependency com.google.api:api-common to v2.1.1 ([#&#8203;533](https://www.togithub.com/googleapis/java-shared-dependencies/issues/533)) ([4e23adb](https://www.github.com/googleapis/java-shared-dependencies/commit/4e23adb666dfaa18ae8bf869aca030153de16d62))
-   update gax.version to v2.7.1 ([#&#8203;544](https://www.togithub.com/googleapis/java-shared-dependencies/issues/544)) ([5a0c480](https://www.github.com/googleapis/java-shared-dependencies/commit/5a0c480671fa834e600edc6ac6fe77967d1b79b3))
-   update google.common-protos.version to v2.7.0 ([#&#8203;542](https://www.togithub.com/googleapis/java-shared-dependencies/issues/542)) ([af97eb5](https://www.github.com/googleapis/java-shared-dependencies/commit/af97eb5d9cff397080fdb3d4a6dbf7c2692e0358))
-   update google.core.version to v2.3.3 ([#&#8203;534](https://www.togithub.com/googleapis/java-shared-dependencies/issues/534)) ([885acef](https://www.github.com/googleapis/java-shared-dependencies/commit/885acefe4ff5313eb53dc43b4825ca597b97dcfd))

</details>

---

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

chore: release 2.0.7 (#253)

:robot: I have created a release \*beep\* \*boop\*
---

* **java:** add -ntp flag to native image testing command ([#1299](https://www.github.com/googleapis/java-retail/issues/1299)) ([#250](https://www.github.com/googleapis/java-retail/issues/250)) ([dd942ab](https://www.github.com/googleapis/java-retail/commit/dd942ab9d8efd582f5c83c259c5ca6e0225c88a5))

* update dependency com.google.cloud:google-cloud-shared-dependencies to v2.6.0 ([#258](https://www.github.com/googleapis/java-retail/issues/258)) ([81f1a09](https://www.github.com/googleapis/java-retail/commit/81f1a0936e779e67d39aa4d68f19b99bea90a819))
---

This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).

chore: release 2.0.8-SNAPSHOT (#260)

:robot: I have created a release \*beep\* \*boop\*
---
---

This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).

chore(owlbot-java): disable re-writing clirr file in owlbot java post-processor (#1320) (#262)

chore: change name to api_shortname in .repo-metadata.json (#263)

chore(java): update variable name to api_shortname in README template (#1323) (#264)

t-karasova committed:
Job_type added to build.sh

sborisenkox committed:
Fix test

sborisenkox committed:
Set up modules configuration.

sborisenkox committed:
Merge remote-tracking branch 'origin/crs-tutorials-search' into crs-tutorials-search

sborisenkox committed:
Merge branch 'main' into crs-tutorials-search

sborisenkox committed:
Merge branch 'googleapis:main' into main

sborisenkox committed:
Merge branch 'main' into crs-tutorials-search

t-karasova committed:
Add 'maven-checkstyle-plugin'

Update pom.xml

Update owlbot.py

sborisenkox committed:
Add 'maven-checkstyle-plugin'

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

sborisenkox committed:
Merge remote-tracking branch 'origin/crs-tutorials-search' into crs-tutorials-search

sborisenkox committed:
Fix test value.

sborisenkox committed:
Fix configuration settings.

sborisenkox committed:
Revert "Fix test value."

This reverts commit 1f9df5c65330ff28371a19ba1a44a104eb86f02c.

kurtisvg commented on .kokoro/build.sh:
```diff
@@ -103,6 +103,30 @@ samples)
         echo "no sample pom.xml found - skipping sample tests"
     fi
     ;;
+tutorials-samples)
```
I think build.sh is a file that's generated by owlbot. If you touch it, it'll revert it back. Do we really need this change?

kurtisvg replied:
I see you've opted out below. What's the reasoning for running this as a separate build as opposed to the rest of the samples?

t-karasova replied:
Please see the Neenu comment below

kurtisvg commented on .kokoro/nightly/tutorials-samples.cfg:
```diff
@@ -0,0 +1,33 @@
+# Format: //devtools/kokoro/config/proto/build.proto
```
Do we need a separate build for tutorials, as opposed to just samples?

t-karasova replied:
We with Neenu have discussed the approach, she suggested the tests to be run with the Nightly Job. So I have added this config and new JOB_TYPE to run just tutorial-samples.

Is that fine or you would like to implement some other approach?

kurtisvg replied:
There is a ["nightly" samples](https://fusion.corp.google.com/projectanalysis/summary/KOKORO/prod:cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-retail%2Fnightly%2Fsamples?search_pattern=cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-retail%2F&project_types=KOKORO&include_inactive_projects=false) build already happening. Do we need a separate job for the tutorials, or can we just keep them all in one job?

Neenu1995 replied:
The reasoning was to keep interactive tutorials separate from the regular samples for ease of code maintenance. Unless there is a strong recommendation against it, let's keep it separate.

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,91 @@
+public class SearchSimpleQuery {
+
+  public static void main(String[] args) throws IOException {
+
+    String projectNumber = System.getenv("PROJECT_NUMBER");
+
+    String defaultCatalogName =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectNumber);
+
+    String defaultSearchPlacementName = defaultCatalogName + "/placements/default_search";
+
```
nit: the empty lines make this hard to read

t-karasova replied:
fixed

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,91 @@
+public class SearchSimpleQuery {
+
+  public static void main(String[] args) throws IOException {
+
```
```suggestion
    // TODO(developer): Replace these variables before running the sample.
```

t-karasova replied:
fixed

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,91 @@
+    search(defaultSearchPlacementName);
+  }
+
+  /**
+   * Call the retail search.
+   *
+   * @return SearchResponse.
+   * @throws IOException if endpoint is not provided.
+   */
```
Please [avoid using java doc for method comments](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#method-comment).

sborisenkox replied:
Changed.

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,91 @@
+   * @throws IOException if endpoint is not provided.
+   */
+  public static SearchResponse search(String defaultSearchPlacementName) throws IOException {
+    // TRY DIFFERENT QUERY PHRASES HERE:
+    String queryPhrase = "Hoodie";
+
+    SearchRequest searchRequest = getSearchRequest(queryPhrase, defaultSearchPlacementName);
+
+    SearchResponse searchResponse =
+        SearchServiceClient.create().search(searchRequest).getPage().getResponse();
+
+    System.out.println("Search response: " + searchResponse);
+
+    return searchResponse;
+  }
+
+  /**
+   * Get search service request.
+   *
+   * @param query search keyword.
+   * @return SearchRequest.
+   */
+  public static SearchRequest getSearchRequest(String query, String defaultSearchPlacementName) {
+    int pageSize = 10;
+
+    String visitorId = UUID.randomUUID().toString();
+
+    SearchRequest searchRequest =
+        SearchRequest.newBuilder()
+            .setPlacement(defaultSearchPlacementName)
+            .setQuery(query)
+            .setVisitorId(visitorId)
+            .setPageSize(pageSize)
+            .build();
+
+    System.out.println("Search request: " + searchRequest);
+
+    return searchRequest;
+  }
+}
```
I'm not really sure why these are two different methods - are they different snippets or should they just be combined? [They also should avoid having a return](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#method-structure).

sborisenkox replied:
Combined two methods in one.

kurtisvg commented on samples/pom.xml:
```diff
@@ -16,9 +16,9 @@
-    <groupId>com.google.cloud.samples</groupId>
-    <artifactId>shared-configuration</artifactId>
-    <version>1.2.0</version>
+    <artifactId>google-cloud-retail-parent</artifactId>
```
Why are you overriding shared-configuration? This should be the parent for all samples

sborisenkox replied:
I have tried to set up configuration because IDE (IntelliJ IDEA) is not recognized Maven modules properly. I can't run my code if Maven modules relations are not configured.

t-karasova replied:
reverted

lesv commented:
# ci/ Units (8)
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running search.SearchWithFacetSpecTest
Search request: placement: "projects/null/locations/global/catalogs/default_catalog/placements/default_search"
query: "Tee"
visitor_id: "284d23e5-6f96-4efd-b57b-95d4e061fba4"
page_size: 10
facet_specs {
  facet_key {
    key: "colorFamilies"
  }
}

Error:  Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 12.411 s <<< FAILURE! - in search.SearchWithFacetSpecTest
Error:  search.SearchWithFacetSpecTest.testSearchWithFacetSpec  Time elapsed: 7.36 s  <<< ERROR!
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at search.SearchWithFacetSpecTest.testSearchWithFacetSpec(SearchWithFacetSpecTest.java:71)

Error:  search.SearchWithFacetSpecTest.testOutput  Time elapsed: 5.027 s  <<< FAILURE!
java.lang.AssertionError
	at search.SearchWithFacetSpecTest.testOutput(SearchWithFacetSpecTest.java:61)

[INFO] Running search.SearchSimpleQueryTest
Search request: placement: "projects/null/locations/global/catalogs/default_catalog/placements/default_search"
query: "Hoodie"
visitor_id: "f9257b3e-e41a-40fe-9583-82e808cf33d9"
page_size: 10

Error:  Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 10.056 s <<< FAILURE! - in search.SearchSimpleQueryTest
Error:  search.SearchSimpleQueryTest.testSearchSimpleQuery  Time elapsed: 4.967 s  <<< ERROR!
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at search.SearchSimpleQueryTest.testSearchSimpleQuery(SearchSimpleQueryTest.java:69)

Error:  search.SearchSimpleQueryTest.testOutput  Time elapsed: 5.085 s  <<< FAILURE!
java.lang.AssertionError
	at search.SearchSimpleQueryTest.testOutput(SearchSimpleQueryTest.java:61)

[INFO] Running search.SearchWithBoostSpecTest
Search request: placement: "projects/null/locations/global/catalogs/default_catalog/placements/default_search"
query: "Tee"
visitor_id: "3ed482de-f9b7-40c5-9da5-0d4a298bc0cc"
page_size: 10
boost_spec {
  condition_boost_specs {
    condition: "(colorFamilies: ANY(\"Blue\"))"
  }
}

Error:  Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 10.236 s <<< FAILURE! - in search.SearchWithBoostSpecTest
Error:  search.SearchWithBoostSpecTest.testSearchWithBoostSpec  Time elapsed: 5.055 s  <<< ERROR!
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at search.SearchWithBoostSpecTest.testSearchWithBoostSpec(SearchWithBoostSpecTest.java:69)

Error:  search.SearchWithBoostSpecTest.testOutput  Time elapsed: 5.176 s  <<< FAILURE!
java.lang.AssertionError
	at search.SearchWithBoostSpecTest.testOutput(SearchWithBoostSpecTest.java:61)

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    SearchSimpleQueryTest.testOutput:61
Error:    SearchWithBoostSpecTest.testOutput:61
Error:    SearchWithFacetSpecTest.testOutput:61
Error:  Errors: 
Error:    SearchSimpleQueryTest.testSearchSimpleQuery:69 » IO The Application Default Cr...
Error:    SearchWithBoostSpecTest.testSearchWithBoostSpec:69 » IO The Application Defaul...
Error:    SearchWithFacetSpecTest.testSearchWithFacetSpec:71 » IO The Application Defaul...
[INFO] 
Error:  Tests run: 6, Failures: 3, Errors: 3, Skipped: 0
```
# ci / dependencies

```
[INFO] --- maven-dependency-plugin:3.1.2:analyze (default-cli) @ retail-install-without-bom ---
Warning:  Unused declared dependencies found:
Warning:     com.google.cloud:google-cloud-retail:jar:2.0.7:compile
Warning:     junit:junit:jar:4.13.2:test
Warning:     com.google.truth:truth:jar:1.1.3:test
```

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Google Cloud Retail Parent 2.0.9-SNAPSHOT .......... SUCCESS [  3.514 s]
[INFO] proto-google-cloud-retail-v2 2.0.9-SNAPSHOT ........ SUCCESS [  0.830 s]
[INFO] grpc-google-cloud-retail-v2 2.0.9-SNAPSHOT ......... SUCCESS [  0.230 s]
[INFO] Google Cloud Retail 2.0.9-SNAPSHOT ................. SUCCESS [  0.492 s]
[INFO] Google Cloud Retail BOM 2.0.9-SNAPSHOT ............. SUCCESS [  0.015 s]
[INFO] Google Cloud Retail Samples Parent 0.0.1-SNAPSHOT .. SUCCESS [  0.087 s]
[INFO] Google Cloud Retail Install Without Bom 0.0.1-SNAPSHOT FAILURE [  0.225 s]
[INFO] Google Cloud Retail Snapshot Samples 0.0.1-SNAPSHOT  SKIPPED
[INFO] Google Cloud Retail Snippets 0.0.1-SNAPSHOT ........ SKIPPED
[INFO] Google Cloud Retail Interactive Tutorials 0.0.1-SNAPSHOT SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

t-karasova committed:
Revert "Revert "Fix test value.""

This reverts commit 886c2eca

tkarasyova committed:
Roll back the configuration settings

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

sborisenkox committed:
Fixes after code review.

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

kurtisvg commented on samples/install-without-bom/pom.xml:
```diff
@@ -11,7 +11,7 @@
     The parent pom defines common style checks and testing strategies for our samples.
     Removing or replacing it should not affect the execution of the samples in anyway.
   -->
-  <parent>
```
nit: this should stay 2 spaces

sborisenkox replied:
Fixed.

kurtisvg commented on samples/interactive-tutorials/pom.xml:
```diff
@@ -0,0 +1,65 @@
+  <name>Google Cloud Retail Interactive Tutorials</name>
+  <url>https://github.com/googleapis/java-retail</url>
+
+ <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.2.0</version>
+  </parent>
```
Please include the [comment here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#project-dependencies).

sborisenkox replied:
Fixed.

kurtisvg commented on samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java:
```diff
@@ -0,0 +1,78 @@
+  }
+
+  public static SearchResponse search(String defaultSearchPlacementName) throws IOException {
+    // TRY DIFFERENT CONDITIONS HERE:
```
nit: it's preferable to link to any documentation on what these values are and what they can do

tkarasyova replied:
We have the tutorials, published in google3, in that tutorials we lead user through the service features description and provide different values which can demonstrate the feature in a particular catalog data

kurtisvg commented on samples/interactive-tutorials/src/test/java/search/SearchSimpleQueryTest.java:
```diff
@@ -0,0 +1,61 @@
+    String defaultCatalogName =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectNumber);
+    defaultSearchPlacementName = defaultCatalogName + "/placements/default_search";
+    Process exec =
+        Runtime.getRuntime()
+            .exec("mvn compile exec:java -Dexec.mainClass=search.SearchSimpleQuery");
+    StreamGobbler streamGobbler = new StreamGobbler(exec.getInputStream());
+    Future<String> stringFuture = Executors.newSingleThreadExecutor().submit(streamGobbler);
+    output = stringFuture.get();
```
What's going on here?

kurtisvg replied:
If you're just trying to get the output, I suggest you try this method: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/kms/src/test/java/kms/SnippetsIT.java#L108-L116

kurtisvg commented on samples/pom.xml:
```diff
@@ -43,6 +45,13 @@
           <skip>true</skip>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <configuration>
+          <skip>true</skip>
+        </configuration>
+      </plugin>
```
Please don't do this. Checkstyle is there for a reason.

t-karasova commented on samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java:
```diff
@@ -0,0 +1,69 @@
+        SearchServiceClient.create().search(searchRequest).getPage().getResponse();
+    System.out.println("Search response: " + searchResponse);
+
+    return searchResponse;
```
We need this return values for tests. In the test we call this method and check the response

t-karasova commented on samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java:
```diff
@@ -0,0 +1,69 @@
+            .build();
+    System.out.println("Search request: " + searchRequest);
+
+    SearchResponse searchResponse =
```
Do you mean we need to put our code (client initialization and client call) into try-catch, like this:

private static SearchResponse getSearchResponse(String query) throws IOException {
    try (SearchServiceClient client = SearchServiceClient.create()) {
      SearchRequest request = SearchRequest.newBuilder().setQuery(query).build();
      SearchResponse response = client.search(request);
      return response;
    }
  }

sborisenkox replied:
Changed.

sborisenkox committed:
Fixes after code review.

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

t-karasova commented:
CLosed due to the CLA issue - the code changes are duplicated to the  PR #289
